qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/BrokerLeaderChangeTest.java

```diff
@@ -82,6 +84,32 @@ public void shouldChangeLeaderAfterLeaderDies() {
     jobCompleter.close();
   }
 
+  @Test
+  public void shouldBeAbleToBecomeLeaderAgain() {
+    // given
+    final var firstLeaderInfo = clusteringRule.getLeaderForPartition(1);
+    final var firstLeaderNodeId = firstLeaderInfo.getNodeId();
+    assertThat(clusteringRule.getCurrentLeaderForPartition(1).getNodeId())
+        .isEqualTo(firstLeaderNodeId);
+
+    // when
+    do {
```

Comment:
Can we comment this? It took me a little a bit to figure out what we're doing - double loop to force leader change until the leader is back again to being the first one :slightly_smiling_face: Clever though :+1: